tests/load/bigquery/test_bigquery_table_builder.py

```diff
@@ -371,3 +380,471 @@ def demo_source() -> DltResource:
+        yield from next(sequence_generator())
+
+    with pytest.raises(
+        ValueError,
+        match=(
+            "are present in both `round_half_away_from_zero` and `round_half_even` "
+            "which is not allowed. They must be mutually exclusive.$"
+        ),
+    ):
+        bigquery_adapter(
+            some_data, round_half_away_from_zero="double_col", round_half_even="double_col"
+        )
+
+
+def test_adapter_hints_parsing_table_description() -> None:
+    @dlt.resource(columns=[{"name": "double_col", "data_type": "double"}])
+    def some_data() -> Iterator[Dict[str, str]]:
+        yield from next(sequence_generator())
+
+    table_description = "Once upon a time a small table got hinted."
+    bigquery_adapter(some_data, table_description=table_description)
+
+    assert some_data._hints["x-bigquery-table-description"] == table_description  # type: ignore
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(all_staging_configs=True, subset=["bigquery"]),
+    ids=lambda x: x.name,
+)
+def test_adapter_hints_table_description(destination_config: DestinationTestConfiguration) -> None:
+    @dlt.resource(columns=[{"name": "col1", "data_type": "text"}])
+    def no_hints() -> Iterator[Dict[str, str]]:
+        yield from [{"col1": str(i)} for i in range(10)]
+
+    hints = bigquery_adapter(
+        no_hints._clone(new_name="hints"),
+        table_description="Once upon a time a small table got hinted.",
+    )
+
+    @dlt.source(max_table_nesting=0)
+    def sources() -> List[DltResource]:
+        return [no_hints, hints]
+
+    pipeline = destination_config.setup_pipeline(
+        f"bigquery_{uniq_id()}",
+        full_refresh=True,
+    )
+
+    # noinspection PyArgumentList
+    pipeline.run(sources())
+
+    with pipeline.sql_client() as c:
+        nc: google.cloud.bigquery.client.Client = c.native_connection
+
+        fqtn_no_hints = c.make_qualified_table_name("no_hints", escape=False)
+        fqtn_hints = c.make_qualified_table_name("hints", escape=False)
+
+        no_hints_table = nc.get_table(fqtn_no_hints)
+        hints_table = nc.get_table(fqtn_hints)
+
+        assert no_hints_table.description is None
+        assert hints_table.description == "Once upon a time a small table got hinted."
+
+
+def test_adapter_hints_parsing_table_expiration() -> None:
+    @dlt.resource(columns=[{"name": "double_col", "data_type": "double"}])
+    def some_data() -> Iterator[Dict[str, str]]:
+        yield from next(sequence_generator())
+
+    bigquery_adapter(some_data, table_expiration_datetime="2030-01-01")
+
+    assert some_data._hints["x-bigquery-table-expiration"] == pendulum.datetime(2030, 1, 1)  # type: ignore
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(all_staging_configs=True, subset=["bigquery"]),
+    ids=lambda x: x.name,
+)
+def test_adapter_hints_table_expiration(destination_config: DestinationTestConfiguration) -> None:
```

Comment:
Is this a valuable hint?

If I should implement it, where can I find a handle to the DDL.

Reply:
hmmm? expiration is super valuable. we may even introduce a standardized hint (not bq specific) for data expiration